server/application/application.go

```diff
@@ -2051,21 +2065,43 @@ func (s *Server) RunResourceAction(ctx context.Context, q *application.ResourceA
 		return nil, fmt.Errorf("error getting Lua resource action: %w", err)
 	}
 
-	newObj, err := luaVM.ExecuteResourceAction(liveObj, action.ActionLua)
+	newObjects, err := luaVM.ExecuteResourceAction(liveObj, action.ActionLua)
 	if err != nil {
 		return nil, fmt.Errorf("error executing Lua resource action: %w", err)
 	}
 
-	newObjBytes, err := json.Marshal(newObj)
-	if err != nil {
-		return nil, fmt.Errorf("error marshaling new object: %w", err)
+	for _, impactedResource := range newObjects {
+		newObj := impactedResource.UnstructuredObj
+		newObjBytes, err := json.Marshal(newObj)
+		if err != nil {
+			return nil, fmt.Errorf("error marshaling new object: %w", err)
+		}
+
+		liveObjBytes, err := json.Marshal(liveObj)
+		if err != nil {
+			return nil, fmt.Errorf("error marshaling live object: %w", err)
+		}
+
+		switch impactedResource.K8SOperation {
+		case "patch":
+			return s.patchResource(ctx, config, liveObjBytes, newObjBytes, newObj)
+		case "create":
+			return s.createResource(ctx, config, app, newObj)
+		default:
+			return nil, fmt.Errorf("unsupported operation: %s", impactedResource.K8SOperation)
+		}
 	}
 
-	liveObjBytes, err := json.Marshal(liveObj)
-	if err != nil {
-		return nil, fmt.Errorf("error marshaling live object: %w", err)
+	if res == nil {
+		s.logAppEvent(a, ctx, argo.EventReasonResourceActionRan, fmt.Sprintf("ran action %s", q.GetAction()))
+	} else {
+		s.logAppEvent(a, ctx, argo.EventReasonResourceActionRan, fmt.Sprintf("ran action %s on resource %s/%s/%s", q.GetAction(), res.Group, res.Kind, res.Name))
+		s.logResourceEvent(res, ctx, argo.EventReasonResourceActionRan, fmt.Sprintf("ran action %s", q.GetAction()))
 	}
+	return &application.ApplicationResponse{}, nil
+}
 
+func (s *Server) patchResource(ctx context.Context, config *rest.Config, liveObjBytes, newObjBytes []byte, newObj *unstructured.Unstructured) (*application.ApplicationResponse, error) {
```

Comment:
I think we need to check whether the resource is allowed in the application, similar to the checks done at the top of `createResource`.

Reply:
Ah sure. I think it would make sense to permit the action to patch resources besides the resource being acted upon. So, for example, if I wanted an action to pause all Workflows under a certain CronWorkflow, I could write an action to do that.

Reply:
But otoh that doesn't have to be part of this PR. Could be saved for a follow-up.

Reply:
Yeah, maybe better to do it as a follow up.
This would mean the non-source object requested for a patch would need to be retrieved first and then the existing merge patch calculation.